rules/integrations/aws/impact_aws_s3_bucket_enumeration_or_brute_force.toml

```diff
@@ -0,0 +1,135 @@
+
+### Response and remediation
+
+- Initiate the incident response process based on the outcome of the triage.
+- Disable or limit the account during the investigation and response.
+- Identify the possible impact of the incident and prioritize accordingly; the following actions can help you gain context:
+    - Identify the account role in the cloud environment.
+    - Assess the criticality of affected services and servers.
+    - Work with your IT team to identify and minimize the impact on users.
+    - Identify if the attacker is moving laterally and compromising other accounts, servers, or services.
+    - Identify any regulatory or legal ramifications related to this activity.
+- Investigate credential exposure on systems compromised or used by the attacker to ensure all compromised accounts are identified. Reset passwords or delete API keys as needed to revoke the attacker's access to the environment. Work with your IT teams to minimize the impact on business operations during these actions.
+- Check if unauthorized new users were created, remove unauthorized new accounts, and request password resets for other IAM users.
+- Consider enabling multi-factor authentication for users.
+- Review the permissions assigned to the implicated user to ensure that the least privilege principle is being followed.
+- Implement security best practices [outlined](https://aws.amazon.com/premiumsupport/knowledge-center/security-best-practices/) by AWS.
+- Take the actions needed to return affected systems, data, or services to their normal operational levels.
+- Identify the initial vector abused by the attacker and take action to prevent reinfection via the same vector.
+- Using the incident response data, update logging and audit policies to improve the mean time to detect (MTTD) and the mean time to respond (MTTR).
+"""
+references = [
+    "https://medium.com/@maciej.pocwierz/how-an-empty-s3-bucket-can-make-your-aws-bill-explode-934a383cb8b1",
+    "https://docs.aws.amazon.com/cli/latest/reference/s3api/"
+]
+risk_score = 21
+rule_id = "5f0234fd-7f21-42af-8391-511d5fd11d5c"
+severity = "low"
+tags = [
+    "Domain: Cloud",
+    "Data Source: AWS",
+    "Data Source: Amazon Web Services",
+    "Resources: Investigation Guide",
+    "Use Case: Log Auditing",
+    "Tactic: Impact"
+]
+timestamp_override = "event.ingested"
+type = "esql"
+
+query = '''
+from logs-aws.cloudtrail*
+| where event.provider == "s3.amazonaws.com" and aws.cloudtrail.error_code == "AccessDenied"
+| stats failed_requests = count(*) by aws.s3.bucket.name, source.address, cloud.account.id
+  // can modify the failed request count or tweak time window to fit environment
+  // can add `not cloud.account.id in (KNOWN)` or specify in exceptions
+| where failed_requests > 40
```

Comment:
no, actually not limiting it is perfect, because then what will happen is there will be an alert for each unique combination of:
* `souce.address`
* `failed_attempts`
* `aws.s3.bucket.name`
* `cloud.account.id`

Which _should_ all be individual alerts.

Remember this is running over 10m windows, looking for 40+ on that combo